File: TWO_SUM/TwoSum.java
import java.util.Scanner;

public class TwoSum {
    public static int[] twoSum(int[] numbers, int target){
        int i = 0, j = numbers.length - 1;

        while (i < j){
            int sum = numbers[i] + numbers[j];

            if (sum == target){
                return new int[]{i, j};
            } else if (sum < target){
                i++;
            }else{
                j--;
            }
        }
        return new int[]{-1, -1};    // safety
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("Enter number of element: ");
        int n = sc.nextInt();

        int[] arr = new int[n];

        System.out.println("Enter sorted array elements: ");
        for (int i = 0; i < n; i++)
            arr[i] = sc.nextInt();

        System.out.println("Enter target: ");
        int target = sc.nextInt();

        int[] result = twoSum(arr, target);

        System.out.println("Indices: " + result[0] + " " + result[1]);

        sc.close();
    }
}

File: two_sum/twosum.c
#include <stdio.h>

void search(int arr[], int n, int X){
	int i = 0, j = n - 1;
	while (i < j){
		int res = arr[i] + arr[j];
		if (res == X){
			printf("%d %d", arr[i], arr[j]);
			return;
		} else if (res < X)
			i++;
		else
			j--;
	}
	printf("No pair found");
}

int main(){
	int n, target;
	printf("Enter size of array: ");
	scanf("%d", &n);
	int arr[n];

	printf("Enter sorted array: ");
	for (int i = 0; i < n; i++) scanf("%d", &arr[i]);

	printf("Enter target: ");
	scanf("%d", &target);

	search(arr, n, target);

	return 0;
}